### PR TITLE
Api/get profile student

### LINKED ISSUE #253: API: Get profile endpoint for student
Create a new endpoint `/api/profile/student` that returns profile information for the student. The information returned should be:
- name
- groupId
- groupName
- timeZone

See instructions on the authentication in #255.

### PR DESCRIPTION
**Pull request by yukafjaponesa**
# Description

create a new endpoint for getting profiles for students,

Fixes #253 

# How to test?
Swagger. 
GET /students/profile
GET /student/profile/{ID}

# Checklist

- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by tpetrina — CHANGES_REQUESTED** [review]
Yuka, you didn't have to create a new table. The student profile is actually in the `users` table. So just change the code to get it from there.

But still, really good job on practicing knex migrations and seeds! 💪

**Review by yukafjaponesa — COMMENTED** [review]
@tpetrina @Elia-Git Elias, you made users api. and i almost did the same. However, im getting error as 500 Internal server Error on swagger. Could you please take a look at my code if it works on your pc? Also, slack seems not working at the moment and I cant conntact on that.

File: src/server/api/routes/api-router.js [hunk 1]
```diff
@@ -56,5 +57,6 @@ router.use('/topics', TopicsRouter);
 router.use('/user', userRouter);
 router.use('/user/register/mentor', mentorRouter);
 router.use('/users', usersRouter);
+router.use('/students_profile', studentsProfileRouter);
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:60]: The URL should be `/profile/student`, not `/students_profile`.

File: src/server/api/routes/studentsprofile.router.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ *         description: The ID of the event to get
+ *
+ *    responses:
+ *      200:
+ *        description: Successful request
+ *      5XX:
+ *        description: Unexpected error.
+ */
+
+router.get('/:id', (req, res, next) => {
```
Comment (tpetrina) [inline src/server/api/routes/studentsprofile.router.js:55]: You don't need this endpoint.

File: src/server/api/routes/studentsprofile.router.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ *    responses:
+ *      200:
+ *        description: Successful request
+ *      5XX:
+ *        description: Unexpected error.
+ */
+
+router.get('/', (req, res, next) => {
+  studentsProfileController
+    .getStudentsProfile()
```
Comment (tpetrina) [inline src/server/api/routes/studentsprofile.router.js:27]: You need to pass id here
```suggestion
    .getStudentsProfile(req.user.id)
```

**Review by tpetrina — CHANGES_REQUESTED** [review]
You need to adjust the code a bit.

File: src/server/api/routes/studentsprofile.router.js (hunk 2 above)
Comment (tpetrina) [reply src/server/api/routes/studentsprofile.router.js:55]: The id is in the `req.user.id`

**Review by yukafjaponesa — COMMENTED** [review]
@tpetrina in swagger, Response body shows
 "select `name`, `group_id`, `groups`.`name` as `group_name`, `timezone` from `users` inner join `groups` on `group_id` = `groups`.`id` inner join `groups` on `group_name` = `groups`.`name` - Not unique table/alias: 'groups'"
which is in studentsprofile.controller.js.
what i chose for leftJoin is wrong or smth is wrong?

**Review by Yaser-Aldammad — APPROVED** [review]

File: src/server/api/routes/studentsprofile.router.js (hunk 3 above)
Comment (Elia-Git) [reply src/server/api/routes/studentsprofile.router.js:27]: Yuka, I think this suggestion also need to be addressed( getStudentsProfile(req.user.id))

**Review by Elia-Git — COMMENTED** [review]
Good job, just a single comment to adresse Toni's Suggestion ( in router on  line 27 )

File: src/server/api/routes/api-router.js [hunk 4]
```diff
@@ -57,6 +57,6 @@ router.use('/topics', TopicsRouter);
 router.use('/user', userRouter);
 router.use('/user/register/mentor', mentorRouter);
 router.use('/users', usersRouter);
-router.use('/students_profile', studentsProfileRouter);
+router.use('/profile/students', studentsProfileRouter);
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:60]:
```suggestion
router.use('/profile/student', studentsProfileRouter);
```

Should be `student` without `s`.

File: src/server/api/controllers/studentsprofile.controller.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const knex = require('../../config/db');
+
+const getStudentsProfile = async () => {
+  try {
+    const studentsProfile = await knex('users')
+      .select('name', 'group_id', 'groups.name as group_name', 'timezone')
```
Comment (tpetrina) [inline src/server/api/controllers/studentsprofile.controller.js:6]:
```suggestion
      .select('name', 'group_id', 'groups.name as groupName', 'timezone')
```

File: src/server/api/controllers/studentsprofile.controller.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const knex = require('../../config/db');
+
+const getStudentsProfile = async () => {
```
Comment (tpetrina) [inline src/server/api/controllers/studentsprofile.controller.js:3]: You aren't using `id` to pick the profile for the logged-in user.

File: src/server/api/controllers/studentsprofile.controller.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  try {
+    const studentsProfile = await knex('users')
+      .select('name', 'group_id', 'groups.name as group_name', 'timezone')
+      .leftJoin('groups', 'group_id', 'groups.id')
+      .leftJoin('groups', 'group_name', 'groups.name');
+    return studentsProfile;
```
Comment (tpetrina) [inline src/server/api/controllers/studentsprofile.controller.js:9]: Is this returning just one profile i.e. for the logged in user? Or is it returning a list of profiles for all users?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Need one more push.

File: src/server/api/controllers/studentsprofile.controller.js (hunk 7 above)
Comment (yukafjaponesa) [reply src/server/api/controllers/studentsprofile.controller.js:9]: nothing returns. only '404 Sorry can't find that!' shows on Response body

Comment (tpetrina) [reply src/server/api/controllers/studentsprofile.controller.js:9]: Well, you need to test it with Postman with instructions from #255.

**Pull request merged**